src/TNetwork.cpp

```diff
@@ -31,6 +31,63 @@
 
 typedef boost::asio::detail::socket_option::integer<SOL_SOCKET, SO_RCVTIMEO> rcv_timeout_option;
 
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <chrono>
+#include <mutex>
+
+std::unordered_map<std::string, std::vector<std::chrono::time_point<std::chrono::high_resolution_clock>>> connectionAttempts;
+std::mutex connectionAttemptsMutex;
+
+bool WatchingConnecting::IsConnectionAllowed(const std::string& clientAddress) {
+        // we check if there is an IP in the blocked list
+        if (WatchingConnecting::IsIPBlocked(clientAddress)) {
+            return false;
+        }
+        std::lock_guard<std::mutex> lock(connectionAttemptsMutex);
+        auto currentTime = std::chrono::high_resolution_clock::now();
+        auto& violations = connectionAttempts[clientAddress];
+
+        // Deleting old violations (older than 5 seconds)
+        violations.erase(std::remove_if(violations.begin(), violations.end(),
```

Comment:
You could use [`erase_if`](https://en.cppreference.com/w/cpp/container/vector/erase2), but that's just a style thing

Reply:
Every change suggested prior to this comment has been worked out so far. Picking up work from here tomorrow!